### PR TITLE
Feature/160451502/create articles

### PR DESCRIPTION
**Pull request by samdiano**
#### What does this PR do?
Creates endpoint to create an article, get an article, update articles and delete articles
#### Description of Task to be completed?
 - add article model
  - add article migration
  - add cloudinary package
  - CRUD for articles
Have these endpoints working.
`GET /api/v1/articles/feed` 
`GET /api/v1/articles/:slug`
`GET /api/v1/articles`
`POST /api/v1/articles`
`PUT /api/v1/articles`
`DELETE /api/v1/articles`

#### How should this be manually tested?
After cloning the repo, cd into it and RUN `npm install`, set env variables, and run migration
- Using Postman, test the endpoints above.
- To test authentication, add `x-access-token: JWT TOKEN`. Get the TOKEN from signup endpoint
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160451502

### CODE REVIEW

**Review by oladimillion — COMMENTED** [review]
LGTM

**Comment by kampkelly** [comment]
LGTM

**Comment by udochukwu** [comment]
LGTM

File: server/controllers/articleController.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+export default class ArticleController {
+  /**
+  * @description -This method creates article for an authenticated user
+  * @param {object} req - The request payload sent from the router
+  * @param {object} res - The response payload sent back from the controller
+  * @returns {object} - status, message and article detail
+  */
+  static createArticle(req, res) {
+    const {
+      title, body, description
```
Comment (onwuvic) [inline server/controllers/articleController.js]: Put it in one line

**Review by slimsolz — COMMENTED** [review]
LGTM

File: server/controllers/articleController.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  * @description -This method gets updates an article
+  * @param {object} req - The request payload sent from the router
+  * @param {object} res - The response payload sent back from the controller
+  * @returns {object} - status, message and articles details
+  */
+  static updateArticle(req, res) {
+    const imgUrl = (req.file ? req.file.secure_url : '');
+    req.body.imgUrl = imgUrl;
+    const request = req.body;
+    Article.update(request, {
```
Comment (s-akinrele) [inline server/controllers/articleController.js]: @samdiano no need to assign a new variable here..

File: server/controllers/articleController.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+  * @description -This method gets an article by the slug
+  * @param {object} req - The request payload sent from the router
+  * @param {object} res - The response payload sent back from the controller
+  * @returns {object} - status, message and list of articles
+  */
+  static getSingleArticle(req, res) {
+    Article.findOne({
+      where: { slug: req.params.slug }
+    }).then(article => (article === null ? res.status(404).json({ message: 'article does not exist', status: 'failed' }) : res.status(200).json({ article })))
```
Comment (s-akinrele) [inline server/controllers/articleController.js]: just use the normal if condition, for consistency across the code base @samdiano

File: server/tests/article.test.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  });
+
+  it('Should create an article', (done) => {
+    chai.request(app)
+      .post('/api/v1/articles')
+      .set('x-access-token', token)
+      .send(article)
+      .end((err, res) => {
+        slug = res.body.article.slug;
+        console.log(res.body);
```
[the remark this answers is not in the document]
Comment (s-akinrele) [reply server/tests/article.test.js]: remove the console.log @samdiano

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**